mbed_host_tests/host_tests_runner/host_test_default.py

```diff
@@ -279,11 +285,16 @@ def start_conn_process():
                             self.logger.prn_wrn("stopped to consume events due to %s event"% key)
                             callbacks_consume = False
                             result = self.RESULT_IO_SERIAL
-                            break
+                            event_queue.put(('__exit_event_queue', 0, time()))
```

Comment:
The point here is that there is now only one exit point for the event queue, which is needed by the bug explained in this PR. This event is used to signal that the queue is "empty" since there is quirk with the `Queue.empty()` function that sometimes returns true erroneously. `callbacks__exit_event_queue` should only be set in one place, otherwise I'd have to replicate that in 3 other events.